sdk/clientcore/core/src/main/java/io/clientcore/core/util/ParameterizedTypeImpl.java

```diff
@@ -15,6 +15,7 @@
 public final class ParameterizedTypeImpl implements ParameterizedType {
```

Comment:
I wonder if this should be named `GenericType` or similar?

Reply:
A list of common generic types in Java includes:
`List<T>`, `Set<T>`, `Map<K, V>`, `Queue<T>`, `Optional<T>`, `Stream<T>`, `Future<T>`, `Callable<T>`, `Comparator<T>`, `Supplier<T>`, `Consumer<T>`, `Function<T, R>`, and etc..

Do we want to support all of those common generic types?

Reply:
How about renaming to a more explicit name: `GenericParametrizedType`?

Reply:
> A list of common generic types in Java include: `List<T>`, `Set<T>`, `Map<K, V>`, `Queue<T>`, `Optional<T>`, `Stream<T>`, `Future<T>`, `Callable<T>`, `Comparator<T>`, `Supplier<T>`, `Consumer<T>`, `Function<T, R>`, and etc..
> 
> Do we want to support all of those common generic types?

Self-answer it from offline: We aiming to support all types. Currently, all types are either .class or ParamertizedType. Such as Map is .class type.